### PR TITLE
Move MDC support to apr-preview version

### PR DESCRIPTION
**Pull request by banibrata**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/online_deployment.py [hunk 1]
```diff
@@ -314,37 +319,27 @@ def _merge_with(self, other: "OnlineDeployment") -> None:
...
-            if len(del_key) > 0:
-                for k in del_key:
-                    del entity.tags[k]
-        else:
-            unflat_data = unflatten(entity.tags, ".")
-            if unflat_data.get("data_collector", None):
-                entity.data_collector = unflat_data.get("data_collector")
-        entity._provisioning_state = deployment.provisioning_state
-        return entity
+    # @classmethod
```
Comment (TajiHarrisMicrosoft) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/online_deployment.py:322]: is there a reason why we're keeping this method commented out instead of deleting?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_ml_client.py [hunk 2]
```diff
@@ -391,7 +391,7 @@ def __init__(
         self._online_endpoints = OnlineEndpointOperations(
             self._operation_scope,
             self._operation_config,
-            self._service_client_02_2022_preview,
+            self._service_client_04_2023_preview,
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_ml_client.py:394]: Do we need to update `online_endpoint_operations` file to reflect this change?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/data_collector.py [hunk 3]
```diff
@@ -2,45 +2,81 @@
...
+    :param sampling_rate: The sampling rate of mdc files, possible values: [0.0, 1.0].
+    :type sampling_rate: float, optional
     :param request_logging: Logging of request payload parameters.
     :type request_logging: RequestLogging, optional
     """
 
     def __init__(
         self,
-        collections: Optional[Dict[str, DeploymentCollection]] = None,
+        *,
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/data_collector.py:29]: if `Collections` isn't optional should it before this line?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/deployment_collection.py [hunk 4]
```diff
@@ -7,23 +7,47 @@
...
-    :param data: Data asset associated with collection logging.
-    :type data: DataAsset, optional
+    :param data: Data asset id associated with collection logging.
+    :type data: str, optional
+    :param client_id: Client ID associated with collection logging.
+    :type client_id: str, optional
 
     """
 
     # pylint: disable=unused-argument,no-self-use
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/deployment_collection.py:25]: nit: we don't need this pylint exception anymore. After the changes

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/online_deployment.py [hunk 5]
```diff
@@ -170,7 +174,7 @@ def __init__(
         :paramtype scoring_script: typing.Optional[typing.Union[str, os.PathLike]]
         """
         self._provisioning_state = kwargs.pop("provisioning_state", None)
-        self.data_collector = kwargs.pop("data_collector", None)
+        # self.data_collector = kwargs.pop("data_collector", None)
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/online_deployment.py:177]: nit: remove this line?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/online_deployment.py [hunk 6]
```diff
@@ -842,10 +835,10 @@ def _from_rest_object(cls, resource: RestOnlineDeploymentData) -> "ManagedOnline
             instance_type=deployment.instance_type,
             endpoint_name=_parse_endpoint_name_from_deployment_id(resource.id),
             instance_count=resource.sku.capacity,
-            private_network_connection=deployment.private_network_connection,
+            # private_network_connection=deployment.private_network_connection,
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/online_deployment.py:838]: nit: Do we need to remove this line or revert this change?
